Sofa/Component/Visual/src/sofa/component/visual/CylinderVisualModel.cpp

```diff
@@ -0,0 +1,93 @@
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/component/visual/CylinderVisualModel.h>
+#include <sofa/core/ObjectFactory.h>
+#include <sofa/core/visual/VisualParams.h>
+#include <sofa/core/ObjectFactory.h>
+
+namespace sofa::component::visual
+{
+
+void registerCylinderVisualModel(sofa::core::ObjectFactory* factory)
+{
+    factory->registerObjects(core::ObjectRegistrationData("A simple visualization for set of cylinder.")
+        .add< CylinderVisualModel >()
+        .addAlias("OglCylinderModel"));
+}
+
+CylinderVisualModel::CylinderVisualModel()
+    : radius(initData(&radius, 1.0f, "radius", "Radius of the cylinder.")),
+      color(initData(&color, sofa::type::RGBAColor::white(), "color", "Color of the cylinders."))
+    , d_edges(initData(&d_edges,"edges","List of edge indices"))
+{
+}
+
+CylinderVisualModel::~CylinderVisualModel() = default;
+
+void CylinderVisualModel::init()
+{
+    VisualModel::init();
+
+    reinit();
+}
+
+void CylinderVisualModel::doDrawVisual(const core::visual::VisualParams* vparams)
+{
+    const VecCoord& pos = this->read( core::ConstVecCoordId::position() )->getValue();
+
+    vparams->drawTool()->setLightingEnabled(true);
+    const float _radius = radius.getValue();
+
+    const sofa::type::RGBAColor& col = color.getValue();
+
+    const SeqEdges& edges = d_edges.getValue();
+
+    for(auto edge : edges)
+    {
+        const Coord& p1 = pos[edge[0]];
+        const Coord& p2 = pos[edge[1]];
+
+        vparams->drawTool()->drawCylinder(p1,p2,_radius,col);
+    }
```

Comment:
Cache draw tools  
```cpp
    auto drawtool = vparams->drawTool();
    drawtool->setLightingEnabled(true)
    for(){
          drawtool->drawCylinder(xxx)
    } 
```
Or even better

    auto drawtool = vparams->drawTool();
    drawtool->setLightingEnabled(true)
    for(){}
    drawtool->drawCylinders(pts);    
```

Reply:
I cached `drawTool`, but the class `DrawTool` does not have `drawCylinders`